discovery/syncer.go

```diff
@@ -492,15 +492,19 @@ func (g *GossipSyncer) Stop() {
 
```

Comment:
while we are at it, could you also stop the iterator in this function here:

```
if !g.isSendingBacklog.CompareAndSwap(false, true) {
		returnSema()
		log.Debugf("GossipSyncer(%x): another goroutine already "+
			"sending backlog, skipping", g.cfg.peerPub[:])

		return nil
	}
```

basically adding a `stop()` at the beginning